app/controllers/web/home.php

```diff
@@ -197,6 +197,35 @@
         ;
     });
 
+App::get('/auth/magic-url')
+    ->groups(['web', 'home'])
+    ->label('permission', 'public')
+    ->label('scope', 'home')
+    ->inject('request')
+    // ->inject('response')
+    ->inject('layout')
+    ->action(function ($request, $layout) {
+        /** @var Utopia\Swoole\Request $request */
+        /** @var Utopia\Swoole\Response $response */
+
+        $page = new View(__DIR__.'/../../views/home/auth/magicURL.phtml');
+        
+        $userId = $request->getQuery('userId');
```

Comment:
This is a huge security bridge. Do not use getQuery with no proper validation. We can use utopia validation. We should also be careful against XSS when printing this vars from the PHTML file. It might be more smart to handle the entire query string completely in client side (inside the inline <script> block).

Reply:
Fixed this now 👍🏻